features/include/pcl/features/impl/scurv.hpp

```diff
@@ -0,0 +1,864 @@
+//
+  h2 = x[2] - x[1];
+  del2 = ( f[2] - f[1] ) / h2;
+//
+//  Set D(1) via non-centered three point formula, adjusted to be
+//  shape preserving.
+//
+  hsum = h1 + h2;
+  w1 = ( h1 + hsum ) / hsum;
+  w2 = -h1 / hsum;
+  d[0] = w1 * del1 + w2 * del2;
+
+  if ( this->pchst ( d[0], del1 ) <= 0.0 )
+  {
+    d[0] = 0.0;
+  }
+//
+//  Need do this check only if monotonicity switches.
+//
+  else if ( this->pchst ( del1, del2 ) < 0.0 )
+  {
+     dmax = 3.0 * del1;
+
+     if ( fabs ( dmax ) < fabs ( d[0] ) )
+     {
+       d[0] = dmax;
+     }
+
+  }
+//
+//  Loop through interior points.
+//
+  for ( i = 2; i <= nless1; i++ )
+  {
+    if ( 2 < i )
+    {
+      h1 = h2;
+      h2 = x[i] - x[i-1];
+      hsum = h1 + h2;
+      del1 = del2;
+      del2 = ( f[i] - f[i-1] ) / h2;
+    }
+//
+//  Set D(I)=0 unless data are strictly monotonic.
+//
+    d[i-1] = 0.0;
+
+    temp = this->pchst ( del1, del2 );
+
+    if ( temp < 0.0 )
+    {
+      ierr = ierr + 1;
+      dsave = del2;
+    }
+//
+//  Count number of changes in direction of monotonicity.
+//
+    else if ( temp == 0.0 )
+    {
+      if ( del2 != 0.0 )
+      {
+        if ( this->pchst ( dsave, del2 ) < 0.0 )
+        {
+          ierr = ierr + 1;
+        }
+        dsave = del2;
+      }
+    }
+//
+//  Use Brodlie modification of Butland formula.
+//
+    else
+    {
+      hsumt3 = 3.0 * hsum;
+      w1 = ( hsum + h1 ) / hsumt3;
+      w2 = ( hsum + h2 ) / hsumt3;
+      dmax = std::max ( fabs ( del1 ), fabs ( del2 ) );
+      dmin = std::min ( fabs ( del1 ), fabs ( del2 ) );
+      drat1 = del1 / dmax;
+      drat2 = del2 / dmax;
+      d[i-1] = dmin / ( w1 * drat1 + w2 * drat2 );
+    }
+  }
+//
+//  Set D(N) via non-centered three point formula, adjusted to be
+//  shape preserving.
+//
+  w1 = -h2 / hsum;
+  w2 = ( h2 + hsum ) / hsum;
+  d[n-1] = w1 * del1 + w2 * del2;
+
+  if ( this->pchst ( d[n-1], del2 ) <= 0.0 )
+  {
+    d[n-1] = 0.0;
+  }
+  else if ( this->pchst ( del1, del2 ) < 0.0 )
+  {
+//
+//  Need do this check only if monotonicity switches.
+//
+    dmax = 3.0 * del2;
+
+    if ( fabs ( dmax ) < fabs ( d[n-1] ) )
+    {
+      d[n-1] = dmax;
+    }
+
+  }
+  return;
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////
+template<typename PointInT, typename PointNT, typename PointOutT> void
+pcl::SCurVEstimation<PointInT, PointNT, PointOutT>::compute (PointCloudOut &output)
+{
+  if (! Feature<PointInT, PointOutT>::initCompute())
+  {
+    output.width = output.height = 0;
+    output.points.clear ();
+    return;
+  }
+  // Copy the header
+  output.header = input_->header;
+
+  // Resize the output dataset
+  // Important! We should only allocate precisely how many elements we will need, otherwise
+  // we risk at pre-allocating too much memory which could lead to bad_alloc 
+  // (see http://dev.pointclouds.org/issues/657)
+  output.width = output.height = 1;
+  output.is_dense = input_->is_dense;
```

Comment:
This likely should be set to true. I assume you can't handle input clouds with nan values, so your descriptor will never generate nans.